server/src/main/java/org/elasticsearch/search/profile/aggregation/AggregationProfileBreakdown.java

```diff
@@ -29,7 +29,10 @@ public AggregationProfileBreakdown() {
      * Add extra debugging information about the aggregation.
      */
     public void addDebugInfo(String key, Object value) {
-        extra.put(key, value);
+        Object old = extra.put(key, value);
+        if (old != null) {
+            throw new IllegalArgumentException("debug info duplicate key [" + key + "] was [" + old + "] is [" + value + "]");
```

Comment:
see, now this is a place where I think an assert would be better than a real exception.  Do we really want to kill a production aggregation because of duplicate debug info?  And if we do, shouldn't it be a 500 class error?  Retrying isn't going to help, right?